exp/views/study.py

```diff
@@ -905,6 +905,18 @@ def dispatch(self, request, *args, **kwargs):
             return HttpResponseRedirect(get_external_url(study, response))
         else:
             _, _, _, study_uuid, _, _, _, *rest = request.path.split("/")
```

Comment:
I think I know the answer to this question, but I would love a better solution.  Is there a less odd way to extract the path directory values?  Could the [`pathlib`](https://docs.python.org/3/library/pathlib.html) library help in anyway?  Feel free to push back on this.  I am fairly sure the answer is "Not worth our time".

Reply:
I had a look at pathLib and that looks like it mostly exists to deal with OS-specific file path issues. It does have the [`parts` method](https://docs.python.org/3/library/pathlib.html#accessing-individual-parts), but that looks like it'll be doing the same thing as `.split("/")` in our case. Let me know if you think there's something I missed.

I'm now thinking regex is the best way to go. I could check whether the request path matches the pattern <root>/studies/<study_UUID>/... or <root>/<locale>/studies/<study_UUID>/..., and do the rest of the manipulation based on that.

Reply:
@okaycj I switched to regex to check the path structure and extract values. Can you please review?